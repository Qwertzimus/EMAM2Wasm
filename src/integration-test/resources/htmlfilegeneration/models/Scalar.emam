package models;

component Scalar {
  ports
    in Q(-10/3 m/s : 10 km/h) inScalar,
    out Q(-10/3 m/s : 10 km/h) outScalar;
}