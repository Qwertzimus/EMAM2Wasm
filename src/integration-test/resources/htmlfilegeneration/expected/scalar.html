<!doctype html>
<html lang="en-us">
<style>
  .model {
    overflow: hidden;
    width: fit-content;
  }

  .ports {
    overflow: hidden;
  }

  .inports {
    float: left;
    margin: 10px;
  }

  .outports {
    float: left;
    margin: 10px;
  }

  .inport-header {
    margin: 0 auto;
    width: fit-content;
  }

  .inport-label {
    float: left;
    margin-right: 5px;
  }

  .inport-field {
    float: right;
    width: 200px;
  }

  .outport-header {
    margin: 0 auto;
    width: fit-content;
  }

  .outport-label {
    float: left;
    margin-right: 5px;
  }

  .outport-field {
    float: left;
    min-width: 200px;
  }

  .execute {
    width: fit-content;
    margin: 10px auto;
  }

  .error {
    color: red;
    margin: 10px;
  }

  .inport-container-inScalar {
    overflow: hidden;
    margin: 3px;
  }

  .outport-container-outScalar {
    overflow: hidden;
    margin: 3px;
  }
</style>
<head>
  <script type="text/javascript"
          src="https://cdnjs.cloudflare.com/ajax/libs/mathjs/4.0.1/math.min.js"></script>
  <script type="text/javascript" src="scalar_wrapper.js"></script>
  <script type="text/javascript" src="scalar.js"></script>
  <title>Model</title>
</head>
<body>
<div class="container">
  <div class="model">
    <div class="ports">
      <div class="inports">
        <div class="inport-header">
          <h3>Inports</h3>
        </div>
        <div class="inport-container-inScalar">
          <div class="inport-label">
            inScalar:
          </div>
          <div class="inport-field">
            <input type="text" id="inport-field-inScalar">
          </div>
        </div>
      </div>
      <div class="outports">
        <div class="outport-header">
          <h3>Outports</h3>
        </div>
        <div class="outport-container-outScalar">
          <div class="outport-label">
            outScalar:
          </div>
          <div class="outport-field" id="outport-field-outScalar"></div>
        </div>
      </div>
    </div>
    <div class="execute">
      <button id="execute" onclick="exec()">Execute</button>
    </div>
  </div>
  <div class="error" id="error"></div>
</div>

<script>
    function exec() {
        document.getElementById("outport-field-outScalar").innerText = "";
        document.getElementById("error").innerHTML = "";

        try {
            setInScalar(document.getElementById("inport-field-inScalar").value);

            execute();

            document.getElementById("outport-field-outScalar").innerText = getOutScalar();
        }
        catch (err) {
            if (err.message === undefined) {
                document.getElementById("error").innerText = err;
            }
            else {
                document.getElementById("error").innerHTML = err.message;
            }
        }
    }
</script>
</body>
</html>
